include/zenoh-pico/api/types.h

```diff
@@ -262,11 +263,11 @@ typedef struct {
  * sent via :c:func:`z_query_reply`.
  *
  * Members:
- *   z_encoding_t encoding: The encoding of the payload.
+ *   z_loaned_encoding_t encoding: The encoding of the payload.
  *   z_attachment_t attachment: an attachment to the response.
  */
 typedef struct {
-    z_encoding_t encoding;
+    z_loaned_encoding_t encoding;
```

Comment:
```suggestion
    z_owned_encoding_t* encoding;
```
here and in other places, see the comment above